src/main/java/uk/gov/hmcts/reform/hmc/config/MessageReceiverConfiguration.java

```diff
@@ -0,0 +1,98 @@
+    private final HearingManagementInterfaceApiClient hmiClient;
+    private static final String MESSAGE_TYPE = "message_type";
+    private static final ObjectMapper OBJECT_MAPPER = new Jackson2ObjectMapperBuilder()
+        .modules(new Jdk8Module())
+        .build();
+
+    public MessageReceiverConfiguration(ApplicationParams applicationParams,
+                                        ActiveDirectoryApiClient activeDirectoryApiClient,
+                                        HearingManagementInterfaceApiClient hmiClient) {
+        this.activeDirectoryApiClient = activeDirectoryApiClient;
+        this.applicationParams = applicationParams;
+        this.hmiClient = hmiClient;
+    }
+
+    @Override
+    @SuppressWarnings("squid:S2189")
+    @PostConstruct
+    public void run() {
+        log.info("Creating Session receiver");
+
+        ServiceBusReceiverClient client = new ServiceBusClientBuilder()
+            .connectionString(applicationParams.getConnectionString())
+            .retryOptions(retryOptions())
+            .receiver()
+            .queueName(applicationParams.getQueueName())
+            .buildClient();
+
+        while (true) {
+            receiveMessages(client);
+        }
+    }
+
+    // handles received messages
+    public void receiveMessages(ServiceBusReceiverClient client) {
+
+        client.receiveMessages(1)
+            .forEach(
+                message -> {
+                    try {
+                        log.info("Received message with id '{}'", message.getMessageId());
+                        MessageType messageType = null;
+                        if (message.getApplicationProperties().containsKey(MESSAGE_TYPE)) {
+                            MessageType.valueOf(message.getApplicationProperties().get(MESSAGE_TYPE).toString());
+                        }
+                        MessageProcessor messageProcessor = new MessageProcessor(applicationParams,
+                                                                                 activeDirectoryApiClient,
+                                                                                 hmiClient);
+                        messageProcessor.processMessage(convertMessage(message.getBody()),
+                                                        messageType, message.getApplicationProperties());
+                        client.complete(message);
+                        log.info("Message with id '{}' handled successfully", message.getMessageId());
+                    } catch (RestClientException ex) {
+                        log.error(ex.getMessage());
+                    } catch (Exception ex) {
+                        log.error(ex.getMessage());
+                    }
+                });
+    }
+
+    private AmqpRetryOptions retryOptions() {
+        AmqpRetryOptions retryOptions = new AmqpRetryOptions();
+        retryOptions.setTryTimeout(Duration.ofSeconds(Integer.valueOf(applicationParams.getWaitToRetryTime())));
```

Comment:
If there's no option for exponential then need to update the LLD

Reply:
sorted to a configurable max int retry approach